NDM: Update profiles documentation (#23374)
DataDog/documentation: content/en/network_monitoring/devices/profiles.md

### sysOID mapped devices

Profiles allow Network Device Monitoring to reuse metric definitions across several device types or instances. Profiles define the metrics to collect as well as how to transform them into Datadog metrics. Each profile is expected to monitor a class of similar devices from the same vendor, they are automatically used by the Datadog Agent by comparing the sysObjectIds of the network device with the one(s) defined in the profile file.

The Datadog Agent is shipped with many out-of-the-box profiles in the `conf.d/snmp.d/default_profiles` directory. This directory is cleaned and reset upon Agent upgrades so do not put anything there. You can write your own custom profiles and extend existing ones by putting files in the `conf.d/snmp.d/profiles` directory

```yaml
sysobjectid:
 - 1.3.6.1.4.1.232.9.4.10
 - 1.3.6.1.4.1.232.9.4.2.*

metrics:
  - MIB: CPQHLTH-MIB
    symbol:
      OID: 1.3.6.1.4.1.232.6.2.8.1.0
      name: cpqHeSysUtilLifeTime
```

This example profile will be used on any network device where its sysobject id either **is** `1.3.6.1.4.1.232.9.4.10` or **starts with** `1.3.6.1.4.1.232.9.4.2.`.

**Advanced**:

> In case you need different metrics for network devices that share the same sysobjectid. You can write profiles without any sysobjectid and force the Datadog Agent to use by configuring the `profile` option in the snmp configuration.

Reviewer: ```suggestion
If you need different metrics for network devices that share the same `sysobjectid`, you can write profiles without any `sysobjectid`, and configure the `profile` option in the SNMP configuration.
```
